### PR TITLE
Added API gateway

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

**Review by BlakeHolifield — APPROVED** [review]
Looks great! Docs and kuttl tests :+1:

File: controllers/cloud.redhat.com/providers/apigateway/provider.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+var ProvName = "apigateway"
+
+// CoreApiGateway is the cronapigateway for the apps apigateways.
+var ApiGatewayConfig = p.NewSingleResourceIdent(ProvName, "apigateway_config", &core.ConfigMap{})
+
+var ApiGatewayDeployment = p.NewSingleResourceIdent(ProvName, "apigateway_deployment", &apps.Deployment{})
+
+var ApiGatewayService = p.NewSingleResourceIdent(ProvName, "apigateway_service", &core.Service{})
+
+// GetEnd returns the correct end provider.
```
[the remark this answers is not in the document]
Comment (psav) [reply controllers/cloud.redhat.com/providers/apigateway/provider.go:19]: bad comment - grr - will fix it up later in my sweep

File: apis/cloud.redhat.com/v1alpha1/clowdenvironment_types.go [hunk 2]
```diff
@@ -232,6 +232,15 @@ type LoggingConfig struct {
 	Mode LoggingMode `json:"mode"`
 }
 
+// ApiGatewayMode just determines if we enable or disable the API gateway
+// +kubebuilder:validation:Enum=enabled;disabled
+type ApiGatewayMode string
+
+// ApiGatewayConfig determines if this env should deploy and API gateway or not
+type ApiGatewayConfig struct {
+	Mode ApiGatewayMode `json:"mode,omitempty"`
```
Comment (kylape) [inline apis/cloud.redhat.com/v1alpha1/clowdenvironment_types.go:241]: let's use web provider mode

File: apis/cloud.redhat.com/v1alpha1/clowdapp_types.go [hunk 3]
```diff
@@ -92,6 +92,10 @@ type PublicWebService struct {
 	// Enabled describes if Clowder should enable the public service and provide the
 	// configuration in the cdappconfig.
 	Enabled bool `json:"enabled,omitempty"`
+
+	// ApiEndpoint is the string that will be used in the api gateway as the services
+	// endpoint.
+	ApiEndpoint string `json:"apiEndpoint,omitempty"`
```
[the remark this answers is not in the document]
Comment (kylape) [reply apis/cloud.redhat.com/v1alpha1/clowdapp_types.go:98]: nah.  also rename to `ApiPrefix`

**Review by kylape — CHANGES_REQUESTED** [review]

**Pull request closed without merging**